src/box/errcode.h

```diff
@@ -115,286 +115,286 @@ struct errcode_record {
+	_(ER_PRIV_NOT_GRANTED, 91,		"No such grant", "user", STRING, "privilege", STRING, "object_type", STRING, "object", STRING) \
+	_(ER_ROLE_NOT_GRANTED, 92,		"User does not have role", "user", STRING, "role", STRING) \
 	_(ER_MISSING_SNAPSHOT, 93,		"Can't find snapshot") \
-	_(ER_CANT_UPDATE_PRIMARY_KEY, 94,	"Attempt to modify a tuple field which is part of primary index in space '%s'") \
-	_(ER_UPDATE_INTEGER_OVERFLOW, 95,	"Integer overflow when performing '%c' operation on field %s") \
+	_(ER_CANT_UPDATE_PRIMARY_KEY, 94,	"Attempt to modify a tuple field which is part of primary index", "space", STRING) \
+	_(ER_UPDATE_INTEGER_OVERFLOW, 95,	"Integer overflow", "operation", STRING, "field", STRING) \
 	_(ER_GUEST_USER_PASSWORD, 96,		"Setting password for guest user has no effect") \
 	_(ER_TRANSACTION_CONFLICT, 97,		"Transaction has been aborted by conflict") \
-	_(ER_UNSUPPORTED_PRIV, 98,		"Unsupported %s privilege '%s'") \
-	_(ER_LOAD_FUNCTION, 99,			"Failed to dynamically load function '%s': %s") \
-	_(ER_FUNCTION_LANGUAGE, 100,		"Unsupported language '%s' specified for function '%s'") \
-	_(ER_RTREE_RECT, 101,			"RTree: %s must be an array with %u (point) or %u (rectangle/box) numeric coordinates") \
-	_(ER_PROC_C, 102,			"%s") \
-	_(ER_UNKNOWN_RTREE_INDEX_DISTANCE_TYPE, 103, "Unknown RTREE index distance type %s") \
-	_(ER_PROTOCOL, 104,			"%s") \
-	_(ER_UPSERT_UNIQUE_SECONDARY_KEY, 105,	"Space %s has a unique secondary index and does not support UPSERT") \
-	_(ER_WRONG_INDEX_RECORD, 106,		"Wrong record in _index space: got {%s}, expected {%s}") \
-	_(ER_WRONG_INDEX_PARTS, 107,		"Wrong index part %u: %s") \
-	_(ER_WRONG_INDEX_OPTIONS, 108,		"Wrong index options: %s") \
-	_(ER_WRONG_SCHEMA_VERSION, 109,		"Wrong schema version, current: %d, in request: %llu") \
-	_(ER_MEMTX_MAX_TUPLE_SIZE, 110,		"Failed to allocate %u bytes for tuple: tuple is too large. Check 'memtx_max_tuple_size' configuration option.") \
-	_(ER_WRONG_SPACE_OPTIONS, 111,		"Wrong space options: %s") \
-	_(ER_UNSUPPORTED_INDEX_FEATURE, 112,	"Index '%s' (%s) of space '%s' (%s) does not support %s") \
-	_(ER_VIEW_IS_RO, 113,			"View '%s' is read-only") \
+	_(ER_UNSUPPORTED_PRIV, 98,		"Unsupported privilege", "object_type", STRING, "privilege", STRING) \
+	_(ER_LOAD_FUNCTION, 99,			"Failed to dynamically load function", "function", STRING, "details", STRING) \
+	_(ER_FUNCTION_LANGUAGE, 100,		"Unsupported function language", "language", STRING, "function", STRING) \
+	_(ER_RTREE_RECT, 101,			"Coordinates number does not make a point or rectangle", "object", STRING, "dimension", UINT) \
+	_(ER_PROC_C, 102,			"Error in C function") \
+	/* ER_UNKNOWN_RTREE_INDEX_DISTANCE_TYPE, 103, Unused */ \
+	_(ER_PROTOCOL, 104,			"Protocol error", "details", STRING) \
+	/* ER_UPSERT_UNIQUE_SECONDARY_KEY, 105,	Unused */ \
+	_(ER_WRONG_INDEX_RECORD, 106,		"Wrong record in _index space", "actual", STRING, "expected", STRING) \
```

Comment:
Better switch to "details".

Reply:
Done.

<details>
<summary>diff</summary>

```diff
diff --git a/src/box/alter.cc b/src/box/alter.cc
index 11df40ec7..ec3bca68f 100644
--- a/src/box/alter.cc
+++ b/src/box/alter.cc
@@ -202,9 +202,10 @@ index_def_check_tuple(struct tuple *tuple)
 		mp_next(&data);
 		p += snprintf(p, e - p, i ? ", %s" : "%s", mp_type_strs[type]);
 	}
-	diag_set(ClientError, ER_WRONG_INDEX_RECORD, got,
-		  "space id (unsigned), index id (unsigned), name (string), "\
-		  "type (string), options (map), parts (array)");
+	diag_set(ClientError, ER_WRONG_INDEX_RECORD,
+		 tt_sprintf("got [%s] tuple while expected "
+		 "[space id (unsigned), index id (unsigned), name (string), "
+		 "type (string), options (map), parts (array)]", got));
 	return -1;
 }
 
diff --git a/src/box/errcode.h b/src/box/errcode.h
index 136887173..afb8eec26 100644
--- a/src/box/errcode.h
+++ b/src/box/errcode.h
@@ -220,7 +220,7 @@ struct errcode_record {
 	/* ER_UNKNOWN_RTREE_INDEX_DISTANCE_TYPE, 103, Unused */ \
 	_(ER_PROTOCOL, 104,			"Protocol error", "details", STRING) \
 	/* ER_UPSERT_UNIQUE_SECONDARY_KEY, 105,	Unused */ \
-	_(ER_WRONG_INDEX_RECORD, 106,		"Wrong record in _index space", "actual", STRING, "expected", STRING) \
+	_(ER_WRONG_INDEX_RECORD, 106,		"Wrong record in _index space", "details", STRING) \
 	_(ER_WRONG_INDEX_PARTS, 107,		"Wrong index part", "partno", UINT, "details", STRING) \
 	_(ER_WRONG_INDEX_OPTIONS, 108,		"Wrong index options", "details", STRING) \
 	_(ER_WRONG_SCHEMA_VERSION, 109,		"Request schema version does not match database schema version", "actual", ULLONG, "expected", ULLONG) \
```

</details>